sqlgen/db_test.go

```diff
@@ -9,6 +9,45 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestPanicOnNoIndex(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("We expect non-indexed queries to panic.")
+		}
+	}()
+
+	tdb, db, err := setup()
+	assert.NoError(t, err)
+	defer tdb.Close()
+
+	db, err = db.WithPanicOnNoIndex()
+
+	// A second set will error
+	_, err = db.WithPanicOnNoIndex()
+	assert.EqualError(t, err, "already is set panic on no index")
+
+	// Querying users without any filters should panic (checked at the top of the test)
+	var users []*User
+	db.Query(context.Background(), &users, nil, nil)
+}
+
+func TestPanicOnNoIndexOverride(t *testing.T) {
+	tdb, db, err := setup()
+	assert.NoError(t, err)
+	defer tdb.Close()
+
+	db, err = db.WithPanicOnNoIndex()
```

Comment:
Is this a safe change to merge into production. Can we safely know we haven't missed a query that doesn't use the index?

Reply:
This is not currently on in production for us. There will be followup PRs where we:
1. only turn this on in test envs (so yes it's safe)
2. in that PR, we will manually tag all full scan queries as such.